lms/djangoapps/student_dashboard/views.py

```diff
@@ -44,16 +65,29 @@ def get_enrolled_past_courses(course_enrollments):
     """
     # TODO move this function out of core code
     enrolled, past = [], []
+    past_course_cards = {}
 
     card_list = get_course_cards_list()
+
     for course in course_enrollments:
         course_card = get_related_card(course.course_overview)
         if course_card in card_list:
             if course.course_overview.has_ended():
-                past.append(course)
+                if course_card.id not in past_course_cards:
+                    past_course_cards[course_card.id] = []
+
+                have_future_course = get_future_courses(course_card.id)
+                if not have_future_course:
+                    course.course_overview.message = "No classes scheduled at the moment"
+
+                past_course_cards[course_card.id].append(course)
             else:
                 enrolled.append(course)
 
+    for card, courses in past_course_cards.items():
+        sorted_courses = sorted(courses, key=lambda course: course.course_overview.start)
+        past.append(sorted_courses[-1])
```

Comment:
please handle index error

Reply:
Loop won't execute if there are not items in it. hence it won't raise index out of bound error.